### PR TITLE
WIP: Saving and loading akira through serialization

### PR DESCRIPTION
**Pull request by albfan**
## Summary / How this PR fixes the problem?

Use existing serialization:

- https://valadoc.org/json-glib-1.0/Json.gobject_serialize.html
- https://valadoc.org/json-glib-1.0/Json.gobject_deserialize.html 

to save and load canvas contents.

## Steps to Test

This is just a PoC so it only saves first item from canvas and loads it as a rectangle.

Example file (you can generate other saving your canvas):

file.akira
```
{"line-width":1.0,"stroke-color-rgba":2863311615,"fill-color-rgba":3435973887,"x":699.6171875,"y":67.75390625,"width":104.0,"height":97.0}
```
- [x] It needs to traverse all items on save
- [x] Seems serialization has problems with CanvasItem (cannot recognize as a type), so it will need to store type (Rect, Ellipse, Text) for each item to deserialize

## Screenshots 
open-save-akira

## Known Issues / Things To Do

- [x] Add akira filter to open/save dialogs
- [x] Test load items order

## This PR fixes/implements the following **bugs/features**:

Relates to #90. A svg format would be preferred anyway.

### CODE REVIEW

**Comment by albfan** [comment]
First functional format:

```
{
  "items": [
    {
      "type": "GooCanvasRect",
      "item": {
        "line-width": 1,
        "stroke-color-rgba": 255,
        "fill-color-rgba": 1318717183,
        "x": 104.32421875,
        "y": 70.765625,
        "width": 282,
        "height": 232
      }
    },
    {
      "type": "GooCanvasEllipse",
      "item": {
        "line-width": 1,
        "stroke-color-rgba": 255,
        "fill-color-rgba": 4012452351,
        "center-x": 368.61328125,
        "center-y": 305.26171875,
        "radius-x": 147.5,
        "radius-y": 145.5,
        "x": 221.11328125,
        "y": 159.76171875,
        "width": 295,
        "height": 291
      }
    }
  ]
}
```

- [ ] Mark objects to not save (as hover and select effect)

**Comment by albfan** [comment]
@AUNaseef please keep in mind this is a tentative design (mostly a PoC that we can discard early), but can you check if you can save/load your draws with this branch?

I would love to load any of your draws and take a look. (Specially for the group feature)

File: src/Services/ActionManager.vala [hunk 1]
```diff
@@ -116,17 +118,156 @@ public class Akira.Services.ActionManager : Object {
     }
 
     private void action_open () {
-        warning ("open");
+        var open_dialog = new Gtk.FileChooserDialog ("Pick a file",
```
Comment (bilelmoussaoui) [inline src/Services/ActionManager.vala]: We need to use a NativeFileChooser here

Comment (albfan) [reply src/Services/ActionManager.vala]: Ok, let me rework it!

**Comment by albfan** [comment]
- [ ] store transformation: Now it is missing. It should be just store and load the transform matrix
- [ ] Group objects. Just store the parent item? It will be done automatically? What if parent do not exists while loading?
- [ ] Mark special objects (like artboards) Includes grouping and intersections (invisible parts for items inside and artboard

**Comment by albfan** [comment]
- [x] Add artboard layers on load

File: src/Services/ActionManager.vala (hunk 1 above)
Comment (albfan) [reply src/Services/ActionManager.vala]: I change into NativeDialog on top of this, but if breaks for me after select a file to open/save. Maybe is due to flatpak. If you can fix it great, by now I will ignore this suggestion, as cannot find any example to implement it.

**Comment by albfan** [comment]
- [x] save json in pretty format

**Pull request closed without merging**